swarm/src/behaviour.rs

```diff
@@ -93,19 +93,22 @@ pub trait NetworkBehaviour: Send + 'static {
     /// The default implementation of this method calls `inject_disconnected` followed with
     /// `inject_connected`. This is a logically safe way to implement this behaviour. However, you
     /// may want to overwrite this method in the situations where this isn't appropriate.
-    fn inject_replaced(&mut self, peer_id: PeerId, closed_endpoint: ConnectedPoint, new_endpoint: ConnectedPoint) {
-        self.inject_disconnected(&peer_id, closed_endpoint);
-        self.inject_connected(peer_id, new_endpoint);
+    #[deprecated(since = "0.6.0", note =
+        "Connections are no longer replaced and this function no longer called. \
+        It will eventually be removed from the trait."
+    )]
+    fn inject_replaced(&mut self, _: PeerId, _: ConnectedPoint, _: ConnectedPoint) {
     }
 
     /// Informs the behaviour about an event generated by the handler dedicated to the peer identified by `peer_id`.
     /// for the behaviour.
     ///
     /// The `peer_id` is guaranteed to be in a connected state. In other words, `inject_connected`
     /// has previously been called with this `PeerId`.
-    fn inject_node_event(
+    fn inject_event(
```

Comment:
I have two minor arguments for this change though:

  1. We have `ConnectionHandler::inject_event` and `ProtocolsHandler::inject_event`. `NetworkBehaviour::inject_event` immediately suggests that it belongs to the same family of functions for communicating events back and forth between a `NetworkBehaviour` and `ProtocolsHandler` / `ConnectionHandler`.
  2. There is no other part of the API left that uses the `node` terminology, since it has been removed in favor of `connection` terminology. It could be `NetworkBehaviour::inject_connection_event`, but again I think the symmetry mentioned in the first point is worth something.